src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -454,12 +359,8 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
             LocalKind::ReturnPointer => {
                 self.not_const();
             }
-            LocalKind::Var if !self.tcx.features().const_let => {
-                if self.mode != Mode::Fn {
-                    emit_feature_err(&self.tcx.sess.parse_sess, "const_let",
-                                    self.span, GateIssue::Language,
-                                    &format!("let bindings in {}s are unstable",self.mode));
-                }
+            LocalKind::Arg |
+            LocalKind::Var if self.mode == Mode::Fn => {
```

Comment:
Question: we used to treat `Arg` differently, right? Is this a bug fix? 

(I'm mainly comparing "before-after" in this PR and this is one of the few places where seem to be doing something different, if you assume that the `const_let` feature gate is enabled)

Reply:
I undid this change. The regular code can handle this just fine. I think this ended up being inserted due to some intermediate change